src/ruby/ext/symengine/extconf.rb

```diff
@@ -0,0 +1,74 @@
+require 'mkmf'
```

Comment:
I am not so sure about the technicalities. I felt the need to use mkmf, from what I saw and read from the tutorials and other projects. Besides it provides many functions for checking and linking libraries. If those can be achieved by cmake, we can get rid of `mkmf`. 

> Does rake install still generate a Makefile? It doesn't seem to generate it on my machine. So it seems we can remove this file. 

I don't think rake generates the makefile. Even without rake, if we simply execute the extconf.rb program by `ruby extconf.rb`, it generates the makefile. Rakefile is just to automate the task.